packages/gatsby-plugin-image/src/components/gatsby-image.browser.tsx

```diff
@@ -50,171 +53,233 @@ export interface IGatsbyImageData {
-  const { width, height, layout, images } = image
-
-  const root = useRef<HTMLElement>()
-  const hydrated = useRef(false)
-  const unobserveRef = useRef<
-    ((element: RefObject<HTMLElement | undefined>) => void) | null
-  >(null)
-  const lazyHydrator = useRef<(() => void) | null>(null)
-  const ref = useRef<HTMLImageElement | undefined>()
-  const [isLoading, toggleIsLoading] = useState(hasNativeLazyLoadSupport())
-  const [isLoaded, toggleIsLoaded] = useState(false)
-
-  if (!global.GATSBY___IMAGE && !hasShownWarning) {
-    hasShownWarning = true
-    console.warn(
-      `[gatsby-plugin-image] You're missing out on some cool performance features. Please add "gatsby-plugin-image" to your gatsby-config.js`
+
+  _lazyHydrate(props, state): Promise<void> {
+    const hasSSRHtml = this.root.current.querySelector(
+      `[data-gatsby-image-ssr]`
     )
+    // On first server hydration do nothing
+    if (hasNativeLazyLoadSupport() && hasSSRHtml && !this.hydrated.current) {
+      this.hydrated.current = true
+      return Promise.resolve()
+    }
+
+    return import(`./lazy-hydrate`).then(({ lazyHydrate }) => {
+      this.lazyHydrator = lazyHydrate(
+        {
+          image: props.image.images,
+          isLoading: state.isLoading,
+          isLoaded: state.isLoaded,
+          toggleIsLoaded: () => {
+            props.onLoad?.()
+
+            this.setState({
+              isLoaded: true,
+            })
+          },
+          ref: this.ref,
+          ...props,
+        },
+        this.root,
+        this.hydrated
+      )
+    })
   }
 
-  const { style: wStyle, className: wClass, ...wrapperProps } = getWrapperProps(
-    width,
-    height,
-    layout
-  )
+  /**
+   * Choose if setupIntersectionObserver should use the image cache or not.
+   */
+  _setupIntersectionObserver(useCache = true): void {
+    import(`./intersection-observer`).then(({ createIntersectionObserver }) => {
+      const intersectionObserver = createIntersectionObserver(() => {
+        if (this.root.current) {
+          const cacheKey = JSON.stringify(this.props.image.images)
+          this.props.onStartLoad?.({
+            wasCached: useCache && hasImageLoaded(cacheKey),
+          })
+          this.setState({
+            isLoading: true,
+            isLoaded: useCache && hasImageLoaded(cacheKey),
+          })
+        }
+      })
+
+      if (this.root.current) {
+        // @ts-ignore - hello
+        this.unobserveRef = intersectionObserver(this.root)
+      }
+    })
+  }
+
+  shouldComponentUpdate(nextProps, nextState): boolean {
+    let hasChanged = false
+
+    // this check mostly means people do not have the correct ref checks in place, we want to reset some state to suppport loading effects
+    if (this.props.image.images !== nextProps.image.images) {
+      // reset state, we'll rely on intersection observer to reload
+      if (this.unobserveRef) {
+        // unregister intersectionObserver
+        this.unobserveRef(this.root)
```

Comment:
This doesn't take an argument. See https://github.com/gatsbyjs/gatsby/blob/master/packages/gatsby-plugin-image/src/components/intersection-observer.ts#L64